esmvaltool/cmorizers/data/formatters/datasets/aeronet.py

```diff
@@ -0,0 +1,412 @@
+        raise ValueError(
+            "Station data contains unexpected number of columns."
+        )
+    return (aod_columns, precipitable_water_columns, angstrom_exponent_columns)
+
+
+def merge_stations(stations):
+    """Collect and merge station data into AeronetStations instance."""
+    columns = {}
+    for name, dtype in (
+        ("station_name", str),
+        ("latitude", np.float64),
+        ("longitude", np.float64),
+        ("elevation", np.float64),
+        ("contacts", str),
+        ("data_frame", object),
+    ):
+        columns[name] = np.array(
+            [getattr(station, name) for station in stations],
+            dtype=dtype,
+        )
+    return AeronetStations(**columns)
+
+
+def assemble_cube(stations, idx, wavelengths=None):
+    """Assemble Iris cube with station data.
+
+    Parameters
+    ----------
+    stations : AeronetStations
+        Station data
+    idx : int
+        Unique ids of all stations
+    wavelengths : list, optional
+        Wavelengths to include in data.
+
+    Returns
+    -------
+    Iris cube
+        Iris cube with station data.
+
+    Raises
+    ------
+    ValueError
+        If station data has inconsistent variable names.
+    """
+    min_time = np.array([df.index.min() for df in stations.data_frame]).min()
+    max_time = np.array([df.index.max() for df in stations.data_frame]).max()
+    date_index = pd.date_range(min_time, max_time, freq="MS")
+    data_frames = [df.reindex(index=date_index) for df in stations.data_frame]
+    all_data_columns = np.unique(
+        np.array([df.columns for df in data_frames], dtype=str),
+        axis=0,
+    )
+    if len(all_data_columns) != 1:
+        raise ValueError(
+            "Station data frames has different sets of column names."
+        )
+    aod_columns, _, _ = sort_data_columns(all_data_columns[0])
+    if wavelengths is None:
+        wavelengths = sorted([int(c[4:-2]) for c in aod_columns])
+
+    aod = da.stack([
+        da.stack([df[f"AOD_{wl}nm"].values for wl in wavelengths], axis=-1)
+        for df in data_frames
+    ], axis=-1)[..., idx]
+    num_days = da.stack([
+        da.stack([
+            df[f"NUM_DAYS[AOD_{wl}nm]"].values.astype(np.float32)
+            for wl in wavelengths
+        ], axis=-1) for df in data_frames
+    ], axis=-1)[..., idx]
+    num_points = da.stack([
+        da.stack([
+            df[f"NUM_POINTS[AOD_{wl}nm]"].values.astype(np.float32)
+            for wl in wavelengths
+        ], axis=-1) for df in data_frames
+    ], axis=-1)[..., idx]
+
+    wavelength_points = da.array(wavelengths, dtype=np.float64)
+    wavelength_coord = iris.coords.DimCoord(
+        points=wavelength_points,
+        standard_name="radiation_wavelength",
+        long_name="Wavelength",
+        var_name="wl",
+        units="nm",
+    )
+    times = date_index.to_pydatetime()
+    time_points = np.array(
+        [datetime(year=t.year, month=t.month, day=15) for t in times])
```

Comment:
Changing this from using numpy to dask (`da.array` instead of `np.array`) raises the following error:
`NotImplementedError: Can not use auto rechunking with object dtype. We are unable to estimate the size in bytes of object data`. This makes sense since it's filled with `datetime` instances in this case. Would it be a problem leaving this particular array as a numpy array, when building the cube?

Reply:
don't think it's a massive issue, it's just one axis :+1: